src/foundry/client-esm/audio/sound.d.mts

```diff
@@ -0,0 +1,401 @@
+   * For a MediaElementAudioSourceNode this simply calls the HTMLAudioElement#pause method directly.
+   */
+  pause(): void;
+
+  /**
+   * Pause playback of the Sound.
+   * This method is factored out so that subclass implementations of Sound can implement alternative behavior.
+   */
+  protected _pause(): void;
+
+  /**
+   * Stop playback for the Sound.
+   * This method is asynchronous because playback may not stop until after an initially provided delay.
+   * The Promise resolves *after* the fade-out of any configured volume transition.
+   * @param options - Options which configure the stopping of sound playback
+   * @returns A Promise which resolves once playback is fully stopped (including fade)
+   */
+  stop(options?: Sound.PlaybackOptions): Promise<this>;
+
+  /**
+   * Stop playback of the Sound.
+   * This method is factored out so that subclass implementations of Sound can implement alternative behavior.
+   */
+  protected _stop(): void;
+
+  /**
+   * Fade the volume for this sound between its current level and a desired target volume.
+   * @param volume  - The desired target volume level between 0 and 1
+   * @param options - Additional options that configure the fade operation
+   * @returns A Promise that resolves after the requested fade duration
+   */
+  fade(
+    volume: number,
+    options?: InexactPartial<{
+      /**
+       * The duration of the fade effect in milliseconds
+       * @defaultValue `1000`
+       */
+      duration: number;
+
+      /**
+       * A volume level to start from, the current volume by default
+       */
+      from: number;
+
+      /**
+       * The type of fade easing, "linear" or "exponential"
+       * @defaultValue `"linear"`
+       */
+      type: "linear" | "exponential";
+    }>,
+  ): Promise<void>;
+
+  /**
+   * Wait a certain scheduled duration within this sound's own AudioContext.
+   * @param duration - The duration to wait in milliseconds
+   * @returns A promise which resolves after the waited duration
+   */
+  wait(duration: number): Promise<void>;
+
+  /**
+   * Schedule a function to occur at the next occurrence of a specific playbackTime for this Sound.
+   * @param fn           - A function that will be called with this Sound as its single argument
+   * @param playbackTime - The desired playback time at which the function should be called
+   * @returns A Promise which resolves to the returned value of the provided function once it has been evaluated.
+   *
+   * @example Schedule audio playback changes
+   * ```js
+   * sound.schedule(() => console.log("Do something exactly 30 seconds into the track"), 30);
+   * sound.schedule(() => console.log("Do something next time the track loops back to the beginning"), 0);
+   * sound.schedule(() => console.log("Do something 5 seconds before the end of the track"), sound.duration - 5);
+   * ```
+   */
+  schedule<R extends Sound.ScheduleCallback>(fn: R, playbackTime: number): Promise<ReturnType<R>>;
```

Comment:
This should be `Promise<Awaited<ReturnType<R>>>`.

So even though the function isn't explicitly awaited, one sometimes underlooked feature of code like this:
```ts
  async schedule(fn, playbackTime) {
      ...
      return fn(this);
     ...
  }
```

Is that any returned promises _will_ be awaited. You can see for yourself:
```js
async function calls(fn) { return fn(); }

console.log(await calls(async () => 1)); // Prints 1, not some nested promise. It's not just a feature of the "await" keyword either.
console.log(calls(async () => 1)); // When not awaited it returns a singly nested promise. Not `Promise<Promise<number>>` just `Promise<number>`.
```